### PR TITLE
Align dragging handle & Style block boundary

### PR DESCRIPTION
**Pull request by attitude**
### Between blocks (overlapping):

Between blocks

### When paragraph is inserted:

Paragraph is inserted

### Above the first block:

Above the first block

### Also adds support for touch devices

localhost_4000_demo_locale=en_donate-page(iPad Air)

Closes https://github.com/contember/private-issues/issues/2
Closes https://github.com/contember/private-issues/issues/37

### Checks
- [x] The changes and implementation strategy have been consulted with the maintainers.
- [x] The code passes all the checks (Run `pnpm run check`).

### CODE REVIEW

**Comment by honzasladek** [comment]
Visually looks 🔝

**Comment by VojtaStanek** [comment]
See:
- Handle next to horizontal line
- Pluses next to horizontal line
- Pluses between blocks
- Maybe too big spacing between text blocks.
- Handle does not work with table well.

image

image


https://user-images.githubusercontent.com/4283010/150508857-6952c306-ff2c-4850-9685-70bc3bd27753.mp4

**Comment by attitude** [comment]
@VojtaStanek thanks for your insightful comment! I had to change quite few pieces to puzzle it together. I've changed the space between block to a minimum.

Here's the preview:

Screenshot 2022-01-22 at 09 49 33

Screenshot 2022-01-22 at 09 49 27

localhost_3000_homepage(iPad Air) (1)

localhost_3000_homepage(Samsung Galaxy S8+)


Cc: @honzasladek

File: packages/ui/src/components/editor/EditorBlockBoundary.tsx [hunk 1]
```diff
@@ -12,12 +14,18 @@ export const EditorBlockBoundary = memo(function EditorBlockBoundary({ blockEdge
...
+				toEnumViewClass(`${blockEdge}Block`),
+			)}
 			contentEditable={false}
 			data-slate-editor={false}
 			onClick={onClick}
 		>
-			<span className={`${prefix}editorBlockBoundary-inner`}>New paragraph</span>
+			<span className={`${prefix}editorBlockBoundary-inner`}>
+				<Icon blueprintIcon="add" />
+				<Label>New paragraph</Label>
```
Comment (VojtaStanek) [inline packages/ui/src/components/editor/EditorBlockBoundary.tsx:27]: quick look: this should be translated

Comment (attitude) [reply packages/ui/src/components/editor/EditorBlockBoundary.tsx:27]: Agree.

The `I18n` is [coming from the `admin` package](https://github.com/contember/admin/blob/pr/editor-block-boundary-and-handle-alignment-style/packages/admin/src/i18n/useMessageFormatter.ts#L80) and non of the UI components use it directly.

Would providing a prop to pass the formatted message be enough?

Comment (VojtaStanek) [reply packages/ui/src/components/editor/EditorBlockBoundary.tsx:27]: Yes, totally.

File: packages/ui/src/components/editor/EditorBlockBoundary.tsx [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+import { } from '../../../src/'
 import { useClassNamePrefix } from '../../auxiliary'
 import { toEnumViewClass } from '../../utils'
+import { Icon } from '../Icon'
+import { Label } from '../Typography/Label'
 
 export interface EditorBlockBoundaryProps {
 	blockEdge: 'before' | 'after'
 	onClick: (e: ReactMouseEvent<HTMLDivElement>) => void
+	newParagraphText?: string
```
Comment (VojtaStanek) [inline packages/ui/src/components/editor/EditorBlockBoundary.tsx:12]:
```suggestion
	newParagraphText: string
```

File: packages/ui/src/components/editor/EditorBlockBoundary.tsx [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+				toEnumViewClass(`${blockEdge}Block`),
+			)}
 			contentEditable={false}
 			data-slate-editor={false}
 			onClick={onClick}
 		>
-			<span className={`${prefix}editorBlockBoundary-inner`}>New paragraph</span>
+			<span className={`${prefix}editorBlockBoundary-inner`}>
+				<Icon blueprintIcon="add" />
+				<Label>{newParagraphText ?? 'New paragraph'}</Label>
```
Comment (VojtaStanek) [inline packages/ui/src/components/editor/EditorBlockBoundary.tsx:29]:
```suggestion
				<Label>{newParagraphText}</Label>
```

File: packages/admin/src/components/bindingFacade/richText/baseEditor/BlockElement.tsx [hunk 4]
```diff
@@ -18,11 +20,14 @@ export const BlockElement = memo(function BlockElement({
...
 	const dataAttributes = ContemberEditor.getElementDataAttributes(element)
 	const El = domElement
+	const formatter = useMessageFormatter(editorDictionary)
+
 	return (
 		<El {...dataAttributes} {...attributes}>
 			{withBoundaries && (
 				<EditorBlockBoundary
 					blockEdge="before"
+					newParagraphText={formatter('editorBlock.editorBlockBoundary.label.newParagraph')}
```
Comment (VojtaStanek) [inline packages/admin/src/components/bindingFacade/richText/baseEditor/BlockElement.tsx:30]: Maybe make it shorter?
```suggestion
					newParagraphText={formatter('editorBlock.editorBlockBoundary.newParagraph')}
```

**Review by VojtaStanek — CHANGES_REQUESTED** [review]

**Review by VojtaStanek — APPROVED** [review]

**Pull request merged**